### ISSUE TITLE
Training loss is not decreasing

### ISSUE THREAD

**Issue by Zailushang211** [tue-mps/eomt#10]
Hi author, nice work.
I tried to finetune with my own dataset without zip into a zip file, just change the dataset loading codes, and have confirmed that the dataset was loaded correctly, but the training loss is not decreasing with the following command, only one instance class in my dataset. Could you provide any suggestion to find the reason?
python3 main.py fit \
  -c configs/coco/instance/eomt_large_640.yaml \
  --trainer.devices 1 \
  --data.batch_size  8 \
  --data.path dataset/RPC \
  --model.ckpt_path models/coco_instance_eomt_large_640/pytorch_model.bin \
  --model.load_ckpt_class_head False

![Image](https://github.com/user-attachments/assets/f7c9b062-5ac3-466d-9408-d5de6386648a)

**Comment by Zailushang211** [comment]
One more thing, the validation step is very very slow, is this normal? one hour for one training epoch, but two hours for val epoch with 5% data.

**Comment by tommiekerssies** [comment]
Hi @Zailushang211, thanks, first of all, when lowering changing the batch size to 8, I would use --trainer.accumulate_grad_batches 2. Secondly, have you specified num_classes to be 1? The validation step for instance segmentation can get slow on machines with poor CPU and high resolution images (beyond >640). This is due to the torchmetrics implementation of AP. You could compute the metric on a lower resolution as a workaround. Let me know if any of this helps.

**Comment by Zailushang211** [comment]
Dear Author, 
thanks for your kindly reply.
To simplify, i changed the the batch size to 4, rainer.accumulate_grad_batches to 1, num_classes is 1, but it makes no difference. About image size, i did not set it myself, just use the 640 instance config yaml.

**Comment by NiccoloCavagnero** [comment]
Hi @Zailushang211, 

Did you try running an evaluation? The loss is already pretty low, similar to the one you would get at convergence on COCO.
Is the classification loss decreasing? As the classification head is the only module without pretraining, I would expect the classification loss to start high and then quickly decrease.

Best, 
Nick

**Comment by Zailushang211** [comment]
Hi Nick, The classification loss is not changing as expected, just similar with the above curves. I tried to inference with the fintuned checkpoints, all  pixels are predicted as backgrounds, while the checkpoints you provided can segment some of the instances.

**Comment by NiccoloCavagnero** [comment]
Could you try finetuning a COCO checkpoint on a default dataset, such as ADE20K? I’m afraid this issue may be related with your own dataset.

Best,
Nick

**Comment by tommiekerssies** [comment]
@Zailushang211 Additionally, could you make sure you're on the latest version of the repository? There was an issue before in our transforms.py which caused images and labels to be misaligned, which has since been fixed, but it's good to verify this is not your issue. If it's indeed not, then indeed please try what @NiccoloCavagnero suggested.

**Comment by Zailushang211** [comment]
Thanks a lot for your help, I  will try Nick's suggestion.
